chainer/function.py

```diff
@@ -299,3 +308,46 @@ def unchain(self):
             if y_ref is not None:
                 y_ref.creator = None
         self.inputs = None
+
+    def add_hook(self, hook, name=None):
+        if name is None:
+            name = repr(hook)
```

Comment:
I change the code how to name function hooks so that it is same as optimizer hooks or extensions in the trainer.